src/esp/nav/PathFinder.cpp

```diff
@@ -597,8 +597,63 @@ struct NavMeshTileHeader {
   dtTileRef tileRef;
   int dataSize;
 };
+
+// Calculate the area of a polygon by iterating over the triangles in the detail
+// mesh and computing their area
+float polyArea(const dtPoly* poly, const dtMeshTile* tile) {
+  float area = 0;
+  // Code to iterate over triangles from here:
+  // https://github.com/recastnavigation/recastnavigation/blob/master/Detour/Source/DetourNavMesh.cpp#L684
+  const unsigned int ip = (unsigned int)(poly - tile->polys);
```

Comment:
Use C++ casts. In this case, you could avoid the cast entirely by using ptrdiff_t..